src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js

```diff
@@ -36,27 +36,40 @@ function rdfResourceSearchDirective($rootScope, $timeout,
       $scope.showInput = function () {
         element.find('.search-rdf-btn')
             .addClass('hidden-xs-up');
-        element.find('.search-rdf-input').css('top', '0px');
+        element.find('.search-rdf-input').css('top', '20px');
         $timeout(function () {
           element.find('.close-rdf-search-btn')
               .removeClass('hidden-xs-up');
           element.find('search-resource-input .view-res-input')
               .focus();
+          window.addEventListener('mousedown', onWindowClick);
         }, 200);
       };
 
       $scope.hideInput = function () {
-        element.find('.search-rdf-input').css('top', '-150px');
-        element.find('search-resource-input .view-res-input')[0].value = '';
-        element.find('search-resource-input .view-res-input').change();
+        element.find('.search-rdf-input').css('top', '-550px');
         element.find('.close-rdf-search-btn')
             .addClass('hidden-xs-up');
         $timeout(function () {
           element.find('.search-rdf-btn')
               .removeClass('hidden-xs-up')
               .css('z-index', '3');
+          element.blur();
+          window.removeEventListener('mousedown', onWindowClick);
         }, 200);
       };
+
+      function onWindowClick(event) {
+        if (!(event.target.id === "search-box" || $(event.target).parents("#search-box").length)) {
```

Comment:
Instead of `parents` you can use `closest`, which is more optimal because it doesn't return all parents, but finds a single anccestor mathing the selector